pyscf/lib/np_helper/imatcopy.c

```diff
@@ -0,0 +1,380 @@
+/* Copyright 2014-2018 The PySCF Developers. All Rights Reserved.
+  
+   Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+ 
+        http://www.apache.org/licenses/LICENSE-2.0
+ 
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ *
+ * Author: Christopher Hillenbrand <chillenbrand15@gmail.com>
+ */
+
+#include <complex.h>
+#include <math.h>
+#include <omp.h>
+#include "np_helper.h"
+
+const int TILESIZE = 32;
+
+#define ELEM_AT(A, i, j, lda) (A)[(i) * (lda) + (j)]
+
+/*
+ * Calculate the largest integer i such that
+ *  i * (i - 1) / 2 <= ijouter.
+ */
+static int uncollapse_loop_index(long long ijouter)
+{
+  return (int) floor((sqrt(0.25 + 2.0 * ijouter) + 0.5));
+}
+
+
+static inline void dtranspose_scale_tile_offdiag(double *A, const int ii, const int jj,
+                                          int lda, const double alpha) {
+  for (int j = jj; j < jj + TILESIZE; j++) {
+#pragma omp simd
+    for(int i = ii; i < ii + TILESIZE; i++) {
+      const double tmp = ELEM_AT(A, i, j, lda);
+      ELEM_AT(A, i, j, lda) = alpha * ELEM_AT(A, j, i, lda);
```

Comment:
1. Remove the macro, simply using `A[i*lda+j]`
2. cast lda to size_t type

Reply:
done!